src/code/index.js

```diff
@@ -88,4 +88,38 @@ function previewLink(button) {
     console.log("Placeholder value:", inputElement.placeholder);
     console.log("Input value:", inputValue);
     // Rest of the code...
-}
\ No newline at end of file
+}
+
+//Function to check if a string is a valid URL
+function isValidURL(url) {
+    try {
+        new URL(url);
+        return true;
+    } catch (error) {
+        return false;
+    }
+}
+
+//Function to handle input change event
+function handleInputChange() {
+    const linkInput = document.getElementById('linkInput');
+    const errorText = document.getElementById('errorText');
+    const previewButton = document.getElementById('previewButton');
+
+    if (isValidURL(linkInput.value)) {
+        //Valid URL
+        linkInput.style.borderBottomColor = 'green';
+        errorText.style.display = 'none';
+        previewButton.classList.add('enabled');
+        previewButton.disabled = false;
+    } else {
+        //Invalid URL
+        linkInput.style.borderBottomColor = 'red';
+        errorText.style.display = 'block';
+        previewButton.classList.remove('enabled');
+        previewButton.disabled = true;
+    }
+}
+
+//Add event listener to the input element
+document.getElementById('linkInput').addEventListener('input', handleInputChange);
```

Comment:
We don't have to set to only a single link input, so we can't use ID, we have to set this event listener to all the input fields in `link-container`.
```html
<li class="link-box">
    <img src="../assets/logos/drag.png" alt="dragger" class="dragger">
    <div class="link">
        <img src="../assets/logos/x.png" alt="discord icon" class="icon">
        <input type="text" placeholder="Paste Social Handle here" />
        <button class="preview" disabled>
            <img src="../assets/logos/share.png" alt="preview link">
        </button>
    </div>
</li>
```
Here, we've a sample of `link-box` container, you can either add a class to the `input` element or iterate through all the `link-box` containers present in `links-container` ul (unordered list) and set function in each of the `input` element it consists.